### ISSUE TITLE
How to get  characteristics by using serviceId

### ISSUE THREAD

**Issue by rohitsangwan01** [woodemi/quick_blue#81]
Hey , as title suggests can we have a method like , discover characteristics

**Comment by Sunbreak** [comment]
Any usecase?

**Comment by rohitsangwan01** [comment]
Yes i was implementing a feature where i have to list out all services and their characteristics
Something similar to nRF connect app
Not urgent feature for now, but just asking in case if its possible
As we are already discovering characterics , so just an api to get list of it, or maybe in discoverServices methods
Instead of getting services , we can get an object containing serviceId and characteristicList

**Comment by Sunbreak** [comment]
Probably. But not on my roadmap

Busy doing pure Dart implementation on Windows

**Comment by rohitsangwan01** [comment]
Okk great , let me know if i can be helpful

**Comment by Sunbreak** [comment]
I'd like to review your implementation on this

**Comment by rohitsangwan01** [comment]
Okk Will try

**Comment by rohitsangwan01** [comment]
@Sunbreak  done , added a PR #82

**Comment by Sunbreak** [comment]
Could split this feature request into another PR?

**Comment by ext-danny** [comment]
Is this api be available already, because I can see this on the example.

**Comment by rohitsangwan01** [comment]
@ext-danny  You can recieve Characteristics from the Service Handler , call discover services and it will return serviceId and Characteristics list

**Comment by ext-danny** [comment]
@rohitsangwan01 I'm using the latest version i.e,   quick_blue: ^0.4.1+1
but the third parameter is missing. if I follow the example like you said I get the error 

**The argument type 'void Function(String, String, List<String>)' can't be assigned to the parameter type 'void Function(String, String)?'.**

checking on the plugin code I can't see the third parameter

**typedef OnServiceDiscovered = void Function(String deviceId, String serviceId);**

**Comment by rohitsangwan01** [comment]
@ext-danny try version 5 , check on Pub dev , its in bera version currently
